src/Microsoft.IdentityModel.Tokens/InMemoryCryptoProviderCache.cs

```diff
@@ -118,22 +158,40 @@ public override bool TryAdd(SignatureProvider signatureProvider)
                 return false;
 
             var cacheKey = GetCacheKey(signatureProvider);
+#if NET461 || NET472 || NETSTANDARD2_0 
+            MemoryCache signatureProviderCache;
+#elif NET45
+            MaxCapacityRefreshCache<string, SignatureProvider> signatureProviderCache;
+#endif
+            // Determine if we are caching a signing or a verifying SignatureProvider.
             if (signatureProvider.WillCreateSignatures)
+                signatureProviderCache = _signingSignatureProviders;
+            else
+                signatureProviderCache = _verifyingSignatureProviders;
+
+#if NET461 || NET472 || NETSTANDARD2_0 
+            var cacheEntryOptions = new MemoryCacheEntryOptions
```

Comment:
nit: cacheEntryOptions could be created only if signatureProvider is not already cached and save some cycles.

Reply:
Good point, fixed. :)